tests/integration/pilot/multicluster_test.go

```diff
@@ -91,6 +95,82 @@ serviceSettings:
 		})
 }
 
+func TestBadRemoteSecret(t *testing.T) {
+	framework.NewTest(t).
+		RequiresMinClusters(2).
+		Features(
+			// TODO tracking topologies as feature labels doesn't make sense
+			"installation.multicluster.multimaster",
+			"installation.multicluster.remote",
+		).
+		Run(func(t framework.TestContext) {
+			// we don't need to test this per-cluster
+			primary := t.Clusters().Configs()[0]
+			// it doesn't matter if the other cluster is a primary/remote/etc.
+			remote := t.Clusters().Exclude(primary)[0]
+
+			var (
+				ns  = i.Settings().SystemNamespace
+				sa  = "istio-reader-no-perms"
+				pod = "istiod-bad-secrets-test"
+			)
+			t.Logf("creating service account %s/%s", ns, sa)
+			if _, err := remote.CoreV1().ServiceAccounts(ns).Create(context.TODO(), &corev1.ServiceAccount{
+				ObjectMeta: metav1.ObjectMeta{Name: sa},
+			}, metav1.CreateOptions{}); err != nil {
+				t.Fatal(err)
+			}
+
+			// intentionally not doing this with subtests since it would be pretty slow
+			for name, opts := range map[string][]string{
```

Comment:
Do we still have unit tests that spin up the full Istiod bootstrap code locally? A bit concerned with this test stability/speed as we are spinning up a new Istiod which is implementation specific, and we have to set the timeout pretty high to not break other tests, while in unit tests we could make it tiny. Not sure if there is a good way to do this though,

Reply:
I don't think we have coverage here (yet). Maybe it's time to add some.

Reply:
```
2021-04-05T20:26:55.702978Z	info	tf	=== DONE (passed):  Test: 'pilot[TestBadRemoteSecret] (17.299965994s)' ===
--- PASS: TestBadRemoteSecret (17.51s)
```

This doesn't seem to bad (for the happy path, assuming no flakiness)